kernel/src/hil/rng.rs

```diff
@@ -66,16 +67,32 @@ pub enum Continue {
     Done,
 }
 
-/// Generic interface for a random number generator
+/// Generic interface for random number generation.
 ///
 /// Implementors should assume the client implements the
-/// [Client](trait.Client.html) trait.
-pub trait RNG {
-    /// Initiate the aquisition of new random number generation.
+/// [Client](trait.Client.html) trait. Note that the random
+/// bits generated do not have entropy guarantees. Therefore,
+/// a caller who needs strong entropy (e.g., for cryptography)
+/// should request many more bits than needed and compress their
+/// entropy with a cryptographic hash and should do so with a
+/// good understanding of the true underlying entropy provided.
+///
+/// Implementers of this
+/// trait should try to ensure that there is at least 0.25 bits of
+/// entropy per generated bit, such that sampling 4x the bits needed
+/// can generate sufficient entropy, but cryptosystems should verify
+/// this is true for each implementation they rely on.
```

Comment:
I'm somewhat ignorant in the details of this field. Is 0.25 per bit a well-known standard that many things provide? Do our current chips meet this requirement?

More generally, "implementors should" is maybe too soft. Do (should?) we want to make some guarantee like "implementations accepted in Tock master are certified by the Tock team to meet this requirement"?

Reply:
For most random number generators I've seen, they provide well above 0.25. The one exception is the MK66, which produces only 1-2 bits for a 32 bit sample. I chose 4 because it seemed like a reasonable tradeoff between caller and callee complexity. I.e., most implementations will not have to whiten their output, and most callers will only need to sample 4x before hashing.

Reply:
In that case, it feels like a line in the sand of 0.25 might encourage a weird duplicated effort of whitening happening both below and above this interface. Should we instead perhaps just report the entropy/bit available from this RNG and let higher layers do with that what they will?

I could then envision a whitening capsule that takes in a bitstream of X entropy/bit and returns one at Y entropy/bit.